source/acquaman/AMScanActionControllerBasicFileWriter.cpp

```diff
@@ -62,6 +65,12 @@ AMScanActionControllerBasicFileWriter::AMScanActionControllerBasicFileWriter(con
 		if(!spectraFile_->open(QIODevice::WriteOnly | QIODevice::Text))
 			emitError(AMScanActionControllerBasicFileWriter::CouldNotOpenError);
 	}
+
+	// start the timer (2mins) to check whether we are writing to the files consistently
+	QTimer *dataFileSizeCheckTimer = new QTimer(this);
```

Comment:
@dretrex  The `dateFileSizeCheckTimer` is a child of `this`. Shouldn't it be destroyed by the parent's destructor? I have thought that in QT, QObject destructor will destroy its children and itself.